core/tests/protractor_utils/general.js

```diff
@@ -256,6 +256,52 @@ var navigateToTopicsAndSkillsDashboardPage = async function() {
   await waitFor.pageToFullyLoad();
 };
 
+var goOffline = async function() {
+  await browser.driver.setNetworkConditions(
+    {
+      offline: true,
+      latency: 0,
+      download_throughput: 0,
+      upload_throughput: 0
+    });
+};
+
+var goOnline = async function() {
+  // Download throughput refers to the maximum number of bytes that can be
+  // downloaded in a given time.
+  // Upload throughput refers to the maximum number of bytes that can be
+  // uploaded in a given time.
+  await browser.driver.setNetworkConditions(
+    {
+      offline: false,
+      latency: 150,
+      download_throughput: 450 * 1024,
+      upload_throughput: 150 * 1024
+    });
+};
+
+var offlineAlert = async function() {
+  await waitFor.visibilityOf(
+    element(by.css('.protractor-test-toast-message')),
+    'Offline warning toast message taking too long to appear.');
+  expect(await element(
+    by.css('.protractor-test-toast-message')).getText()).toMatch(
+    'Looks like you are offline. You can continue working, and can save ' +
+    'your changes once reconnected.');
+  await waitFor.invisibilityOf(
+    element(by.css('.protractor-test-toast-message')),
+    'Offline warning toast message taking too long to appear.');
+};
+
+var onlineAlert = async function() {
```

Comment:
ditto: move to page utility file
ditto: name as verb

Reply:
Done.